### PR TITLE
Added check for `/` DimensionMismatch error

### LINKED ISSUE #369: Error message for `/` is reversed
Consider:

```
A = rand(3,2)
B = rand(4,5)
```

`A\B` correctly gives:

> LoadError: DimensionMismatch("left hand side has 3 rows, but right hand side has 4 rows")
> while loading ..., in expression starting on line ...

But: `A/B` gives:

> LoadError: DimensionMismatch("left hand side has 5 rows, but right hand side has 2 rows")
> while loading ... in expression starting on line ...

Which is incorrect. 

The correct error message would be:

> LoadError: DimensionMismatch("left hand side has 2 columns, but right hand side has 5 columns")
> while loading ..., in expression starting on line ...

That is because it is dispatching to `\`, 
and not catching, then rethrowing the error after reversing

---

​I suggest that `DimentionMismatch` should be parameter-ised to have a `lhs::Int` and `rhs::Int` field, and a `indexname::Union{String,Int}`, and a suitable `display` method; or something kind of like that.

---

(editted: previous version had my corrected output in the quote for what was supposed to be the current (incorrect) output, and visa-versa)

### LINKED ISSUE #18742: Error message for `/` is reversed
Consider:

```
A = rand(3,2)
B = rand(4,5)
```

`A\B` correctly gives:

> LoadError: DimensionMismatch("left hand side has 3 rows, but right hand side has 4 rows")
> while loading ..., in expression starting on line ...

But: `A/B` gives:

> LoadError: DimensionMismatch("left hand side has 5 rows, but right hand side has 2 rows")
> while loading ... in expression starting on line ...

Which is incorrect. 

The correct error message would be:

> LoadError: DimensionMismatch("left hand side has 2 columns, but right hand side has 5 columns")
> while loading ..., in expression starting on line ...

That is because it is dispatching to `\`, 
and not catching, then rethrowing the error after reversing

---

​I suggest that `DimentionMismatch` should be parameter-ised to have a `lhs::Int` and `rhs::Int` field, and a `indexname::Union{String,Int}`, and a suitable `display` method; or something kind of like that.

---

(editted: previous version had my corrected output in the quote for what was supposed to be the current (incorrect) output, and visa-versa)

### PR DESCRIPTION
**Pull request by raghav9-97**
Fixes JuliaLang/LinearAlgebra.jl#369

### CODE REVIEW

**Comment by raghav9-97** [comment]
Please review this PR

**Comment by ViralBShah** [comment]
Needs a better error message. This one is a bit confusing.

**Comment by raghav9-97** [comment]
@ViralBShah What would be a good error message?

**Comment by ViralBShah** [comment]
How about: `Both inputs should have the same number of columns`. I think it is useful for the error message to say what needs to be fixed. Let's apply similar language update for the `\' error message too.

Also the code could be made compact:
```
size(A,2) != size(B,2) && throw(DimensionMismatch("Both inputs should have the same number of columns"))
return copy(adjoint(adjoint(B) \ adjoint(A)))
```

Also, we only really need one test per code path for this, so maybe only one test each for `/` and `\`.

**Comment by raghav9-97** [comment]
Fine, I will update the error message for both ```/``` ```\```

**Comment by raghav9-97** [comment]
Is this PR now ready for merging?

**Comment by andreasnoack** [comment]
> Also, we only really need one test per code path for this, so maybe only one test each for `/` and `\`.

@ViralBShah I'm not completely sure what you are suggesting here but if it's just to test this in `\` then that is what we are already doing. The issue in #30446 is that this might be confusing so adding an extra check would be more user-friendly although technically redundant.

A minor comment about the error message. I'm not a native speaker so I might be wrong here but in Danish *both* together with *the same* would be pleonastic. I.e. it should either be "Both inputs should have $(size(A,2)) number of columns" or "The inputs should have the same number of columns".

**Comment by ViralBShah** [comment]
@andreasnoack My only point was that we don't need so many tests to test for this error. Just one would suffice.

**Comment by ViralBShah** [comment]
I think the current error phrase sounds better to me, even though it is pleonastic (Thank you - I learnt a new word from a non-native speaker :-)

**Comment by raghav9-97** [comment]
Thanks @ViralBShah for suggesting a good error message and @andreasnoack for reviewing the code.

File: stdlib/LinearAlgebra/src/qr.jl [hunk 1]
```diff
@@ -854,7 +854,7 @@ function (\)(A::Union{QR{TA},QRCompactWY{TA},QRPivoted{TA}}, B::AbstractVecOrMat
     @assert !has_offset_axes(B)
     S = promote_type(TA,TB)
     m, n = size(A)
-    m == size(B,1) || throw(DimensionMismatch("left hand side has $m rows, but right hand side has $(size(B,1)) rows"))
+    m == size(B,1) || throw(DimensionMismatch("Both inputs should have the same number of rows"))
```
Comment (fredrikekre) [inline stdlib/LinearAlgebra/src/qr.jl]: Would be good to have kept the values, isn't that valuable information?

**Pull request merged**